src/components/ui/LoginForm.tsx

```diff
@@ -1,89 +1,64 @@
 'use client';
 import { Amplify } from 'aws-amplify';
-import {
-  AuthUser,
-  fetchAuthSession,
-  getCurrentUser,
-  signInWithRedirect,
-  signOut,
-} from 'aws-amplify/auth';
-import { Hub } from 'aws-amplify/utils';
+import { fetchAuthSession, signInWithRedirect } from 'aws-amplify/auth';
+import { useRouter } from 'next/navigation';
 import { useEffect, useState } from 'react';
 
-const OAUTH_DOMAIN: string = process.env.NEXT_PUBLIC_OAUTH_DOMAIN || '';
-const LOCALHOST_URL: string = process.env.NEXT_PUBLIC_LOCALHOST_URL || '';
-const OAUTH_REDIRECT_URL: string = process.env.NEXT_PUBLIC_OAUTH_REDIRECT_URL || '';
-const USER_POOL_ID: string = process.env.NEXT_PUBLIC_USER_POOL_ID || '';
-const USER_POOL_CLIENT_ID: string = process.env.NEXT_PUBLIC_USER_POOL_CLIENT_ID || '';
-Amplify.configure({
-  Auth: {
-    Cognito: {
-      loginWith: {
-        oauth: {
-          domain: OAUTH_DOMAIN, // OAuth domain
-          scopes: ['openid'], // Scope needed
-          responseType: 'code',
-          redirectSignIn: [LOCALHOST_URL, OAUTH_REDIRECT_URL],
-          redirectSignOut: [LOCALHOST_URL, OAUTH_REDIRECT_URL],
+const OAUTH_DOMAIN = process.env.NEXT_PUBLIC_OAUTH_DOMAIN || '';
+const LOCALHOST_URL = process.env.NEXT_PUBLIC_LOCALHOST_URL || '';
+const OAUTH_REDIRECT_URL = process.env.NEXT_PUBLIC_OAUTH_REDIRECT_URL || '';
+const USER_POOL_ID = process.env.NEXT_PUBLIC_USER_POOL_ID || '';
+const USER_POOL_CLIENT_ID = process.env.NEXT_PUBLIC_USER_POOL_CLIENT_ID || '';
+
+Amplify.configure(
+  {
+    Auth: {
+      Cognito: {
+        loginWith: {
+          oauth: {
+            domain: OAUTH_DOMAIN,
+            scopes: ['openid'],
+            responseType: 'code',
+            redirectSignIn: [LOCALHOST_URL + '/login'],
+            redirectSignOut: [LOCALHOST_URL, OAUTH_REDIRECT_URL],
+          },
         },
+        userPoolId: USER_POOL_ID,
+        userPoolClientId: USER_POOL_CLIENT_ID,
       },
-      userPoolId: USER_POOL_ID,
-      userPoolClientId: USER_POOL_CLIENT_ID,
     },
   },
-});
+  { ssr: true },
```

Comment:
Reference: https://docs.amplify.aws/react/build-a-backend/server-side-rendering/nextjs/#configure-amplify-library-for-client-side-usage

> NOTE: To use the Amplify library on the client side in a Next.js app, you will need to set ssr to true when calling Amplify.configure. This instructs the Amplify library to store tokens in the cookie store of a browser. Cookies will be sent along with requests to your Next.js server for authentication.